doc/doxygen/headers/automatic_and_symbolic_differentiation.h

```diff
@@ -14,47 +14,389 @@
+ *    specialized based on the specific inputs (possibly carrying a differential history) given to that specific
+ *    operator on the vertex of the DAG. In this way, a compile-time optimized set of instructions can be generated
+ *    for the very specialized individual operations used to evaluate each intermediate result of the dependent
+ *    function.
+ *
+ * Each of these methods, of course, has its advantages and disadvantages, and one may be more appropriate
+ * than another for a given problem that is to be solved. As the aforemetioned implementational details
+ * (and others not discussed) may be hidden from the user, it may still be important to understand the
+ * implications, run-time cost,  and potential limitations, of using any one of these "black-box"
+ * auto-differentiable numbers.
+ *
+ * Resources used to furnish the details supplied here include:
+ *
+ * @code{.bib}
+ * @InProceedings{Fike2011a,
+ *   author    = {Fike, Jeffrey A and Alonso, Juan J},
+ *   title     = {The Development of Hyper-Dual Numbers for Exact Second-Derivative Calculations},
+ *   booktitle = {49th {AIAA} Aerospace Sciences Meeting including the New Horizons Forum and Aerospace Exposition},
+ *   year      = {2011},
+ *   volume    = {886},
+ *   pages     = {124},
+ *   month     = {jan},
+ *   publisher = {American Institute of Aeronautics and Astronautics},
+ *   doi       = {10.2514/6.2011-886},
+ * }
+ * @endcode
+ *
+ * @code{.bib}
+ * @Manual{Walther2009a,
+ *   title     = {Getting Started with ADOL-C},
+ *   author    = {Walther, Andrea and Griewank, Andreas},
+ *   year      = {2009},
+ *   booktitle = {Combinatorial scientific computing},
+ *   doi       = {10.1.1.210.4834},
+ *   pages     = {181--202}
+ * }
+ * @endcode
+ *
+ * ### Explotation of the chain-rule
+ *
+ * In the most practical sense, any of the above categories exploit the chain-rule to compute the total
+ * derivative of a composite function. To perform this action, they typically use one of two mechanisms to
+ * compute derivatives, specifically
+ * - <em>forward-mode</em> (or <em>forward accumulation</em>) auto-differentation, or
+ * - <em>reverse-mode</em> (or <em>reverse accumulation</em>) auto-differentation.
+ * As a point of interest, the <em>optimal Jacobian accumulation</em>, which performs a minimal set of
+ * computations, lies somewhere between these two limiting cases. Its computation for a general composite
+ * function remains an open problem in graph theory.
+ *
+ * With the aid of the diagram below (it and some of the listed details courtesy of this
+ * <a href="https://en.wikipedia.org/wiki/Automatic_differentiation">Wikipedia article</a>),
+ *
+ * <div class="twocolumn" style="width: 80%">
+ *   <div class="parent">
+ *     <div class="img" align="center">
+ *       <img src="https://upload.wikimedia.org/wikipedia/commons/a/a4/ForwardAccumulationAutomaticDifferentiation.png"
```

Comment:
These two images
```
https://commons.wikimedia.org/wiki/File:ForwardAccumulationAutomaticDifferentiation.png
https://commons.wikimedia.org/wiki/File:ReverseaccumulationAD.png
```
have been released into the public domain. Does this then mean that we can host copies ourselves that can be downloaded with the offline documentation?

Reply:
I would just keep the link and not bother with hosting it ourselves.